plugins/org.eclipse.glsp.ide.editor/src/org/eclipse/glsp/ide/editor/utils/UIUtil.java

```diff
@@ -17,18 +17,25 @@
 
 import java.util.Optional;
 
+import org.apache.log4j.Logger;
+import org.eclipse.glsp.ide.editor.ui.GLSPEditorIntegrationPlugin;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
 import org.eclipse.ui.IEditorPart;
 import org.eclipse.ui.IWorkbench;
 import org.eclipse.ui.IWorkbenchPage;
 import org.eclipse.ui.IWorkbenchWindow;
 import org.eclipse.ui.PlatformUI;
 
 public final class UIUtil {
```

Comment:
Can we avoid promoting usage of magic/static ways to retrieve editor-specific contexts?

While it makes sense for the Display (Display.getDefault() being also static itself), I don't really like these:

- findShell() (As it's only a fallback, I would make it private)
- getActiveEditor() (This one is only used in an AbstractHandler, which already provides info about the Event; it can give access to the proper Editor instance without any magic. Moreover, the EclipseActionHandler.findContext() method is never actually used, so I would get rid of it entirely)

Reply:
All good points, I removed all unnecessary methods and made findShell private until someone actually needs it.